plugins/flatpak/gs-flatpak.c

```diff
@@ -3678,6 +3679,76 @@ app_has_local_source (GsApp *app)
 		(url != NULL && g_str_has_prefix (url, "file://"));
 }
 
+static guint64
+get_installation_dir_free_space (GsFlatpak *self, GError **error)
+{
+	g_autoptr (GFile) installation_dir = NULL;
+	g_autoptr (GFileInfo) info = NULL;
+
+	installation_dir = flatpak_installation_get_path (self->installation);
+
+	info = g_file_query_filesystem_info (installation_dir,
+					     G_FILE_ATTRIBUTE_FILESYSTEM_FREE,
+					     NULL,
+					     error);
+	if (info == NULL)
+		return 0;
```

Comment:
It doesn't necessarily need a change, because in both call sites, we check the value of the GError * rather than the return value, but perhaps as some stylistic advice it's better to have an error return value which isn't a valid return value - either G_MAXUINT64 or make the function into a gboolean and have the space filled into a guint64 *. Because in typical glib (@pwithnall can correct me if I am wrong!), GError ** may be NULL, and this means that a caller then can't distinguish an error (whatever it is) from there being literally 0 free space - and/or if someone did if (return value == 0) to detect an error, *error would be unset if there was no error (rather than genuinely no space), potentially causing a SEGV in the calling code.

Reply:
> @pwithnall can correct me if I am wrong!

You are right. FWIW, I’d prefer returning the `guint64` as an out parameter, rather than as a return value with `g_assert (error != NULL)` because, while adding the `g_assert()` is correct, it’s an anti-pattern for public APIs. So I don’t really want people to ever think it’s acceptable.